test/compilable/issue15574.sh

```diff
@@ -46,9 +43,7 @@ int main() {
 }
 EOF
 
-if [ -z ${CC+x} ]; then CC=cc; fi
-
-$CC -m${MODEL} -c -o ${C_FILE}${OBJ} $C_FILE
+cc -m${MODEL} -c -o ${C_FILE}${OBJ} $C_FILE
```

Comment:
So we managed to reach the state where CC refers to the C++ compiler:

https://github.com/dlang/dmd/blob/cce909b19a1e6b03a182d2a1d488841e2cc64b53/test/tools/d_do_test.d#L588-L596

https://github.com/dlang/dmd/blob/cce909b19a1e6b03a182d2a1d488841e2cc64b53/test/README.md#L188

But in this case CC is used as people normally use it: the C compiler.

In this case it makes a difference whether a C or C++ compiler is used.
I would suggest we introduce a new environment variable (e.g. `CPP`) and let CC point to the system's C compiler.
FWIW in the build make scripts the C++ compiler is at least called `HOST_CXX`.